File: simple-pos-management-app/server/app/schemas/product.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

from app.schemas.category import Category


class ProductBase(BaseModel):
    name: str
    sku: str
    price: float = Field(gt=0)
    stock_quantity: int = Field(ge=0)
    description: Optional[str] = None
    category_id: int
    status: bool = True


class ProductCreate(ProductBase):
    pass


class Product(ProductBase):
    id: int
    created_at: datetime
    updated_at: datetime
    category: Category

    class Config:
        from_attributes = True

File: simple-pos-management-app/client/src/types/product.ts
import { Category } from "./category";

export interface ProductBase {
  name: string;
  sku: string;
  price: number;
  stock_quantity: number;
  description?: string;
  category_id: number;
  status: boolean;
}

export interface Product extends ProductBase {
  id: number;
  created_at: string;
  updated_at: string;
  category: Category;
}

export type ProductCreate = ProductBase;

export interface ProductPaginationResponse {
  items: Product[];
  total_pages: number;
  total: number;
}
